### ISSUE TITLE
OPA library to return what the same object that OPA returns

### ISSUE THREAD

**Issue by antonioberben** [solo-io/gloo-mesh#1951]
@jmunozro has the full context if needed

### Version

1.2.x (latest stable)

### Is your feature request related to a problem? Please describe.

Currently OPA functionality does not return to the envoy filters what OPA server returns.

### Describe the solution you'd like

Using OPA as a standalone server, the decision making process returns an object such as:
```
{ allowed: false, body: "HTTP verb not allowed" }
```

The customer would like to retrieve that object to be able to parse it later in another envoy filter.

This task consist of two parts:
1. Make our OPA integration, I think it is [here](https://github.com/solo-io/ext-auth-service/blob/c94a78ee914fb4b497d25146a7bafcc08eeb25b4/pkg/config/opa/config.go#L87) to return the object that OPA library generates.
2. Make that object available for the other filters. i.e. Use transformation filter to return `403` with payload being the body returned by OPA (from the example above)

In summary, they need to be able to handle the reason OPA has failed
